### PR TITLE
James 2788 - close threads in Unit tests

### PR DESCRIPTION
**Pull request by remk**

### CODE REVIEW

**Comment by remk** [comment]
test this please

File: server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java [hunk 1]
```diff
@@ -101,22 +102,25 @@ void listShouldReturnTaskDetailsWhenTaskInProgress() throws Exception {
             .body("[0].status", is(TaskManager.Status.IN_PROGRESS.getValue()))
             .body("[0].taskId", is(taskId.asString()))
             .body("[0].class", is(not(empty())));
+
+        waitingForResultLatch.countDown();
```
Comment (mbaechler) [inline server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java]: There's a problem with this solution: a failing test still leak resources.
What about handling the cleanup (here, calling countDown) in some @AfterEach method or writing an extension to manage the CountDownLatch?

**Comment by remk** [comment]
test this please

**Comment by trantienduchn** [comment]
```
[a62d6c03959e8bd85410d14ed1131230fc1ab0ab] [ERROR] Errors: 
[a62d6c03959e8bd85410d14ed1131230fc1ab0ab] [ERROR]   ForwardSmtpTest.forwardingAnEmailShouldWork:79 » ConditionTimeout Assertion co...
```
this test is blocking CI builds.

**Comment by trantienduchn** [comment]
test this please

**Comment by trantienduchn** [comment]
```
[a62d6c03959e8bd85410d14ed1131230fc1ab0ab] [ERROR] Errors: 
[a62d6c03959e8bd85410d14ed1131230fc1ab0ab] [ERROR]   ForwardSmtpTest.forwardingAnEmailShouldWork:79 » ConditionTimeout Assertion co...
```

**Comment by remk** [comment]
> ```
> [a62d6c03959e8bd85410d14ed1131230fc1ab0ab] [ERROR] Errors: 
> [a62d6c03959e8bd85410d14ed1131230fc1ab0ab] [ERROR]   ForwardSmtpTest.forwardingAnEmailShouldWork:79 » ConditionTimeout Assertion co...
> ```

The AwsS3RabbitMQForwardSmtpTest implementation doesn't terminate 100% of the time on my local machine.

**Comment by remk** [comment]
test this please

**Comment by mbaechler** [comment]
```
[ERROR]   RabbitMQEventBusTest$LifeCycleTest$SingleEventBus$DispatchingWhenNetWorkIssue.dispatchShouldWorkAfterNetworkIssuesForOldRegistration:333 » ConditionTimeout
```
test this please

**Comment by remk** [comment]
`[ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:3.4.4:add-source (scala-compile-first) on project apache-james-mailbox-event-json: Execution scala-compile-first of goal net.alchim31.maven:scala-maven-plugin:3.4.4:add-source failed: Plugin net.alchim31.maven:scala-maven-plugin:3.4.4 or one of its dependencies could not be resolved: Could not transfer artifact org.scala-lang:scala-library:jar:2.10.6 from/to central (https://repo.maven.apache.org/maven2): GET request of: org/scala-lang/scala-library/2.10.6/scala-library-2.10.6.jar from central failed: Connection reset -> [Help 1]`

network failure ?

**Comment by remk** [comment]
test this please

**Comment by trantienduchn** [comment]
```
[a62d6c03959e8bd85410d14ed1131230fc1ab0ab] + cd dockerfiles/run/guice/memory
[a62d6c03959e8bd85410d14ed1131230fc1ab0ab] + docker cp james-parent-test-james-2788-0ddffef8-04d6-4b5b-8a61-79066441f7a2:null destination
[a62d6c03959e8bd85410d14ed1131230fc1ab0ab] Error response from daemon: Could not find the file null in container james-parent-test-james-2788-0ddffef8-04d6-4b5b-8a61-79066441f7a2
```
try to rebase?

**Comment by remk** [comment]
test this please

**Comment by remk** [comment]
no more logs access.
test this please

**Comment by rouazana** [comment]
merged

**Pull request closed without merging**